CI, BUG: Tests Failing on Python 3.9

[Recent CI run](https://github.com/Becksteinlab/kda/actions/runs/6062024919/job/16447826664) failed on Python 3.9. 

It appears there is an issue with `test_kda.py::test_get_ordered_cycle_all_node_cycles` (output pasted below) where a cycle is now considered ambiguous based on the original diagram. 

```
FAILED kda/tests/test_kda.py::test_get_ordered_cycle_all_node_cycles - kda.exceptions.CycleError: Distinct ordered cycle could not be determined. Input diagram has multiple unique cycles that contain all nodes in the input cycle ([0, 4, 3, 2, 1]). To fix ambiguity, please input a cycle with the nodes in the correct orientation. Select one of the following possibilities: [[0, 1, 2, 3, 4], [0, 1, 2, 4, 3], [0, 1, 3, 2, 4], [0, 1, 3, 4, 2], [0, 1, 4, 2, 3], [0, 1, 4, 3, 2], [0, 2, 1, 3, 4], [0, 2, 1, 4, 3], [0, 2, 3, 1, 4], [0, 2, 4, 1, 3], [0, 3, 1, 2, 4], [0, 3, 2, 1, 4]]
```

I'm guessing this is due to a change in the `NetworkX` library. This should be fixed before addressing #70.

--- a/kda/tests/test_kda.py
+++ b/kda/tests/test_kda.py
@@ -4,7 +4,6 @@
 #
 # Kinetic Diagram Analysis Testing
 
-import sys
 import pytest
 import numpy as np
 from numpy.testing import assert_almost_equal, assert_allclose, assert_array_equal
@@ -1191,7 +1190,10 @@ def test_get_ordered_cycle_all_node_cycles():
     G = nx.MultiDiGraph()
     graph_utils.generate_edges(G, K)
 
-    if sys.version_info < (3, 9):
+    nx_version = nx.__version__
+    major = int(nx_version[0])
+    minor = int(nx_version[2])
+    if (major, minor) < (3, 1):
         valid_cycle = [0, 4, 3, 2, 1]
         invalid_cycle = [0, 1, 2, 3, 4]
     else:

--- a/kda/tests/test_kda.py
+++ b/kda/tests/test_kda.py
@@ -4,6 +4,7 @@
 #
 # Kinetic Diagram Analysis Testing
 
+import sys
 import pytest
 import numpy as np
 from numpy.testing import assert_almost_equal, assert_allclose, assert_array_equal
@@ -1189,14 +1190,24 @@ def test_get_ordered_cycle_all_node_cycles():
     # generate the diagram and edges
     G = nx.MultiDiGraph()
     graph_utils.generate_edges(G, K)
+
+    if sys.version_info < (3, 9):
+        valid_cycle = [0, 4, 3, 2, 1]
+        invalid_cycle = [0, 1, 2, 3, 4]
+    else:
+        # order must be changed due to updates in
+        # `nx.simple_cycles()`. For more details
+        # see https://github.com/Becksteinlab/kda/issues/71
+        valid_cycle = [0, 1, 2, 3, 4]
+        invalid_cycle = [0, 4, 3, 2, 1]
     # run for a valid all-node cycle
-    calculations._get_ordered_cycle(G, [0, 4, 3, 2, 1])
+    calculations._get_ordered_cycle(G, valid_cycle)
     # run for case where there are more nodes than possible in a cycle
     with pytest.raises(CycleError):
         calculations._get_ordered_cycle(G, [0, 1, 2, 3, 4, 5])
     # run for an all-node case where the order is incorrect
     with pytest.raises(CycleError):
-        calculations._get_ordered_cycle(G, [0, 1, 2, 3, 4])
+        calculations._get_ordered_cycle(G, invalid_cycle)
 
 
 def test_function_inputs():

